src/lib.rs

```diff
@@ -0,0 +1,8 @@
+mod grain;
+mod matrix;
+mod permutation;
+mod poseidon;
+mod spec;
+
+pub use crate::poseidon::Poseidon;
+pub use crate::spec::{MDSMatrices, MDSMatrix, SparseMDSMatrix, Spec, State};
```

Comment:
I think that if the only thing required to be exposed is `Poseidon`, and its fields are private or `pub(crate)` there's no need to export anything from `spec`.

Reply:
@CPerezz um but transcript circuit side some of them are helpful

Reply:
Should be resolved with [f287bf6](https://github.com/privacy-scaling-explorations/poseidon/pull/2/commits/f287bf6bcf4b76ac7e9010de98e707569ee90223)